profiles/common/modules/custom/ecas/includes/ecas.inc

```diff
@@ -354,7 +369,29 @@ function ecas_sync_user_info(&$user, array $user_info, array $args) {
   // Let the opportunity to modules to alter user roles.
   drupal_alter('ecas_user_roles', $edit['roles'], $user, $user_info, $args);
 
+  // We just gave the possibility to others modules to treat the case, now we
```

Comment:
Function `ecas_sync_user_info` is also used in `ecas_import_users_show_user_content_form_submit` and in that case we don't want to redirect the user.

If a login comes without an associated email we should reject the login and don't sync any info, so we shouldn't even get here in this code.

Reply:
I will find a better way.

I was not aware about the submit callback. I place the test there to let sites altering the process for there specific use cases. So, they can alter the platform behaviour without twisting the platform.